lib/spells/BattleSpellMechanics.cpp

```diff
@@ -445,16 +445,21 @@ void BattleSpellMechanics::beforeCast(BattleSpellCast & sc, vstd::RNG & rng, con
 
 	std::vector <const battle::Unit *> resisted;
 
-	auto filterResisted = [&, this](const battle::Unit * unit) -> bool
+	resistantUnitIds.clear();
+	if(isNegativeSpell() && isMagicalEffect())
 	{
-		if(isNegativeSpell() && isMagicalEffect())
+		//magic resistance
+		for (const auto * unit : battle()->battleGetAllUnits(false))
```

Comment:
Would've been great to avoid testing resistance for every unit on battlefield for every cast of every spell, and only perform it if unit is actually hit by the spell...

Don't see any clear way to do this though. Any ideas?

Reply:
Let's keep as it for now.

Performance is very unlikely to be a factor here, it just feels illogical to test for resistance when resistance likely won't even happen.